outputs.tf

```diff
@@ -31,3 +31,13 @@ output "private_ip_addresses" {
     for key, pe in azurerm_private_endpoint.this : key => pe.private_service_connection[0].private_ip_address
   }
 }
+
+output "principal_id" {
+  description = "The Principal ID associated with this Managed Service Identity."
+  value       = azurerm_cognitive_account.this.identity[0].principal_id
```

Comment:
@lonegunmanb Question: is it correct to create an output for each element of the identity block of the resource ? or I should have created a single output exposing the all block ?

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cognitive_account

Reply:
@zioproto I think we need a `try` here since it's possible to have an account without `identity` block.

>is it correct to create an output for each element of the identity block of the resource ? or I should have created a single output exposing the all block ?

Both works for me, I would say let's output the identity block:

```hcl
output "cognitive_account_identity" {
  ...
  value = try(azurerm_cognitive_account.this.identity[0], null)
}